src/main/java/com/javadiscord/jdi/internal/gateway/handlers/events/codec/handlers/guild/message/MessageDeleteHandler.java

```diff
@@ -7,5 +7,7 @@
 
 public class MessageDeleteHandler implements EventHandler<Message> {
     @Override
-    public void handle(Message event, ConnectionMediator connectionMediator, Discord discord) {}
+    public void handle(Message event, ConnectionMediator connectionMediator, Discord discord) {
+        discord.getCache().getCacheForGuild(event.guildId()).remove(event.id(), event.getClass());
```

Comment:
```suggestion
        discord.getCache().getCacheForGuild(event.guildId()).remove(event.id(), event);
```
also remove from all other places

Reply:
Can't do this, we use `Class` as the type

Reply:
but ur doing a `.getClass()` on usage site as wel

Reply:
```java
public void remove(long id, Class<?> clazz) {
        models.remove(toId(id, clazz));
    }
```

```java
default String toId(long id, Class<?> clazz) {
        return id + "-" + clazz.getName();
    }
```
The effect here does not matter.